core_utilities/archive_checker.py

```diff
@@ -0,0 +1,451 @@
+"""Application for checking the status of videos in the pony archive."""
+
+import tkinter as tk
+import csv
+import threading
+import requests
+import math
+import io
+import asyncio
+import aiohttp
+import json
+import os
+
+from tkinter import filedialog, Event, ttk
+from PIL import ImageTk, Image
+from typing import List, Tuple
+from yt_dlp import YoutubeDL, DownloadError
+from enum import Enum
+from classes.gui import GUI
+
+class ArchiveIndices:
+    LINK = 3
+    TITLE = 4
+    CHANNEL = 5
+    STATE = 7
+    ALT_LINK = 8
+    FOUND = 9
+    NOTES = 10
+
+class States(Enum):
+    NON_EMBEDDABLE = ('non-embedable', 'non-embeddable')
+    UNAVAILABLE = ('unavailable', 'deleted', 'private', 'tos deleted')
+    AGE_RESTRICTED = ('age-restricted',)
+    BLOCKED = ('blocked',)
+
+    @classmethod
+    def get(cls, value: str):
+        for state in cls:
+            if value.lower() in state.value:
+                return state
+
+blocked_everywhere_indicator = 'EVERYWHERE EXCEPT:'
+video_not_found = "Video not found"
+lock = asyncio.Lock()
+
+
+class ArchiveStatusChecker(GUI):
+    def __init__(self):
+        super().__init__()
+        self.running = False
+        self.output_csv_path = ""
+        self.checking_range = []
+        self.processed_videos = 0
+        self.starting_row_num = 2
+        self.async_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.async_loop)
+
+    def gui(self, root):
+
+        # Only get archive or check for updates when it's not being checked so full progress
+        # label and output path is displayed upon revisiting
+        if not self.running:     
+            csv_str: str = requests.get('https://docs.google.com/spreadsheets/d/1rEofPkliKppvttd8pEX8H6DtSljlfmQLdFR-SlyyX7E/export?format=csv').content.decode()
```

Comment:
Make the archive URL a class variable or global so that it's easier to change and reuse